src/components/input/TypeHandler/TypeHandler.ts

```diff
@@ -27,6 +27,10 @@ export abstract class TypeHandler {
 	protected constructor(protected readonly component: Component<any>) {
 		this.value = this.component.value
 	}
+	onBlur(event: FocusEvent) {
+		this.state = this.handleBlur(this.state, event)
+	}
+	abstract handleBlur(state: State, event: FocusEvent): State
```

Comment:
This should not be abstract but rather have a default implementation so that implementors of the class don't need to bother providing a stub every time. Also change to `StateEditor` for the type of the first argument and remove `event`.